testing/travis-fork-check.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import logging
+import os
+import subprocess
+import sys
+
+def check_travis_repo_slug():
+    if 'encrypted_6a9d32f3e0bd_key' not in os.environ:
+        logging.warning("Decryption disabled. Setting MANTL_CI_FORK_CHECK to 1")
+        print("1")
+    else:
+        logging.info("Decrypting OS ssh key")
```

Comment:
So that's an ordering issue, right? It's too tightly coupled now. Not reason for it to be in the same script, when a script run after could do the same thing and be easier to discover the functionality.

> On May 11, 2016, at 11:03 AM, Sam Hatfield <notifications@github.com> wrote:
> 
> In testing/travis-fork-check.py:
> 
> > @@ -0,0 +1,30 @@
> > +#!/usr/bin/env python
> > +from __future__ import print_function
> > +import logging
> > +import os
> > +import subprocess
> > +import sys
> > +
> > +def check_travis_repo_slug():
> > +    if 'encrypted_6a9d32f3e0bd_key' not in os.environ:
> > +        logging.warning("Decryption disabled. Setting MANTL_CI_FORK_CHECK to 1")
> > +        print("1")
> > +    else:
> > +        logging.info("Decrypting OS ssh key")
> Take a look at this error’d build from Travis: https://travis-ci.org/CiscoCloud/mantl/jobs/127433850 In this version, the `openssl` command was run as its own step in `.travis.yml`, but it would error out on PRs from forks. So my solution was to wrap that around a check to see if the encrypted vars were actually available. If they are, then run the decryption, otherwise, skip it.
> …
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
>